src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseIndexOperatorDiagnosticAnalyzer.cs

```diff
@@ -201,25 +189,28 @@ protected override void InitializeWorker(AnalysisContext context)
             // Also ensure that the left side of the subtraction : `s.Length - value` is actually
             // getting the length off the same instance we're indexing into.
 
-            lengthLikePropertyOpt ??= TryGetLengthLikeProperty(infoCache, targetMethodOpt);
-            if (lengthLikePropertyOpt == null ||
-                !IsInstanceLengthCheck(lengthLikePropertyOpt, instance, subtraction.LeftOperand))
+            lengthLikeProperty ??= TryGetLengthLikeProperty(infoCache, targetMethod);
+            if (lengthLikeProperty == null ||
+                !IsInstanceLengthCheck(lengthLikeProperty, instance, subtraction.LeftOperand))
             {
                 return;
             }
 
+            if (CSharpSemanticFacts.Instance.IsInExpressionTree(instance.SemanticModel, instance.Syntax, infoCache.ExpressionOfTType, cancellationToken))
```

Comment:
Nit: Could move up. No point checking for a length property, options, etc. etc.

Reply:
those are generally the cheap things to do.  WE effectively just check it off the iops given us.  no need to do the complex binding checks about expression trees (the rare case).